geometry-builder.cpp

```diff
@@ -760,8 +759,71 @@ void geometry_builder::set_exclude_broken_polygon(int exclude)
     excludepoly = exclude;
 }
 
+void geometry_builder::set_reprojection(struct reprojection *r)
+{
+    reprojection = r;
+}
+
+/**
+ * Computes area of given polygonal geometry.
+ * \return the area in projected units, or in EPSG 3857 if area reprojection is enabled
+ */
+double geometry_builder::getArea(const geos::geom::Geometry *geom) const
+{
+    // reprojection is not necessary, or has not been asked for.
+    if (!reprojection) {
+        return geom->getArea();
+    }
+
+    // MultiPolygon - return sum of individual areas
+    if (const geos::geom::MultiPolygon* multi = dynamic_cast<const geos::geom::MultiPolygon *>(geom)) {
+        double area = 0.0;
+        for (std::size_t i=0; i<multi->getNumGeometries(); i++) {
+            area += getArea(multi->getGeometryN(i));
+        }
+        return area;
```

Comment:
In this particular case, a loop would mean considerably less writing:

    for (auto g : multi) {
      area += getArea(g);
    }

Personally, I find that more readable. However, in general, using the standard library function and lambdas in place of hand-crafted for loops is a good thing. So sure, go ahead and use it.

(Also: please use auto inside the if condition.)